sql/src/main/java/org/apache/druid/sql/calcite/planner/IngestHandler.java

```diff
@@ -106,12 +110,45 @@ protected String operationName()
   @Override
   public void validate()
   {
-    if (ingestNode().getPartitionedBy() == null) {
+    if (ingestNode().getTargetTable() instanceof ExternalDestinationSqlIdentifier) {
+      if (!handlerContext.plannerContext().featureAvailable(EngineFeature.WRITE_EXTERNAL_DATA)) {
```

Comment:
Is this validation necessary? `IngestHandler` is only supported via the MSQ engine, therefore will this message show up? And if so, let's add a test for the same.

Reply:
IngestHandler is used by native queries as well, just that it throws an exception immediately. Currently, this wouldn't be triggered since the child classes of IngestHandler already check if it is an insert or replace.

However, for correctness's sake, this should be present, even if a test cannot reach it. If in the future, say, a new engine is added which supports insert and replace, but not export, this would be needed.